src/Files.App/Actions/Show/ToggleSidebarAction.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) 2024 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+namespace Files.App.Actions
+{
+	internal sealed class ToggleSidebarAction : ObservableObject, IToggleAction
+	{
+		private readonly ISidebarViewModel SidebarViewModel = Ioc.Default.GetRequiredService<SidebarViewModel>();
+
+		public string Label
+			=> "ToggleSidebar".GetLocalizedResource();
+
+		public string Description
+			=> "ToggleSidebarDescription".GetLocalizedResource();
+
+		public HotKey HotKey
+			=> new(Keys.S, KeyModifiers.CtrlAlt);
+
+		public bool IsOn =>
+			SidebarViewModel.SidebarDisplayMode is SidebarDisplayMode.Expanded
+				? true
+				: false;
+
+		public ToggleSidebarAction()
+		{
+			SidebarViewModel.PropertyChanged += ViewModel_PropertyChanged;
+		}
+
+		public Task ExecuteAsync(object? parameter = null)
+		{
+			SidebarViewModel.SidebarDisplayMode = IsOn
```

Comment:
Maybe we need this?
```suggestion
			SidebarViewModel.SidebarDisplayMode = !IsOn
```

Reply:
SidebarDisplayMode isn't a bool

Reply:
Yeah after this line there's "?!" statement.

Reply:
Integers 0..2 from an enum.
https://github.com/files-community/Files/blob/566cc8cee12c71da9d352233cf7cb2b84dae9ab9/src/Files.App/UserControls/Sidebar/SidebarDisplayMode.cs#L9-L23

Reply:
Try with and without "!"